### PR TITLE
Updated fight caves plugin

### PR DESCRIPTION
**Pull request by DrizzyBot**
Adds a new overlay that tells you the current wave you're on. Also tells you what monsters are remaining on your current wave, as well as monsters coming up in the next wave (if toggled on).

### CODE REVIEW

**Comment by jplsek** [comment]
For the future, please try to avoid opening a new pull request for the same feature/bugfix. You could have rebased your previous PR's branch.

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveConfig.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.fightcave;
+
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+@ConfigGroup
+		(
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveConfig.java]: One too many tabs on all the annotation description

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveConfig.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		keyName = "fightcaves",
+		name = "Fight Caves",
+		description = "Configuration for the Fight Caves plugin"
+		)
+public interface FightCaveConfig extends Config
+{
+	@ConfigItem
+			(
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveConfig.java]: One too many tabs

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveConfig.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+			keyName = "showNextWave",
+			name = "Show next wave",
+			description = "Shows monsters that will be spawning on the wave after your current one."
+			)
+
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveConfig.java]: Remove new line

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, Devin French <https://github.com/devinfrench>
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java]: Do not remove copyright unless you are the original copyright holder. You add copyright.
It appears that git is just being git and this is actually a completely new file with the old name. Maybe renaming this file to `FightCaveWaveOverlay` would be good to clear up confusion and facilitate future expansion.

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, Devin French <https://github.com/devinfrench>
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: Don't remove copyright

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 6]
```diff
@@ -40,6 +46,8 @@
 @PluginDescriptor(
 	name = "Fight Cave"
 )
+
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: No blank line between annotations

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java [hunk 7]
```diff
@@ -27,99 +27,205 @@
...
+	public Dimension render(Graphics2D graphics)
 	{
-		if (protectFromMissilesImg == null)
+		int currentWave = plugin.getCurrentWave();
+		if (currentWave == 0)
 		{
-			String path = "/prayers/protect_from_missiles.png";
-			protectFromMissilesImg = getImage(path);
+			return null;
 		}
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java]: Blank line between logical blocks (after the closing brace)

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java [hunk 8]
```diff
@@ -27,99 +27,205 @@
...
-
-	private BufferedImage getImage(String path)
-	{
-		BufferedImage image = null;
-		try
+		panelComponent.getChildren().clear();
+		panelComponent.getChildren().add(TitleComponent.builder()
+				.text("Wave " + currentWave)
+				.color(Color.orange)
+				.build());
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java]: Blank line after logical blocks

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java [hunk 9]
```diff
@@ -27,99 +27,205 @@
...
+			if (quantity <= 0)
 			{
-				image = ImageIO.read(FightCaveOverlay.class.getResourceAsStream(path));
+				continue;
 			}
+			panelComponent.getChildren().add(TitleComponent.builder()
+					.text(quantity + "x " + monsters.get(monsterID))
+					.color(Color.white)
+					.build());
 		}
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java]: Blank line after logical blocks

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 10]
```diff
@@ -51,25 +59,102 @@
...
 	@Override
-	public Overlay getOverlay()
+	public Collection<Overlay> getOverlays()
 	{
-		return overlay;
+		return Arrays.asList(overlay, jadOverlay);
 	}
 
 	@Schedule(
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: This should be above `update` method

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 11]
```diff
@@ -51,25 +59,102 @@
...
 	)
+
+	@Provides
+	FightCaveConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(FightCaveConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: `startUp` is not required to be overridden and can be removed

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 12]
```diff
@@ -51,25 +59,102 @@
...
+		return Arrays.equals(client.getMapRegions(), FIGHT_CAVE_REGION);
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() != ChatMessageType.SERVER)
+		{
+			return;
+		}
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: Blank line after

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 13]
```diff
@@ -51,25 +59,102 @@
...
+	}
+
+	@Subscribe
+	public void onActorDeath(ActorDeath event)
+	{
+		Actor actor = event.getActor();
+		if (currentWave == 0)
+		{
+			return;
+		}
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: Blank line after

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java [hunk 14]
```diff
@@ -51,25 +59,102 @@
...
+	}
+
 	public void update()
 	{
 		if (client.getGameState() != GameState.LOGGED_IN)
 		{
 			return;
 		}
-
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCavePlugin.java]: Keep blank line

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/JadOverlay.java [hunk 15]
```diff
@@ -0,0 +1,100 @@
+package net.runelite.client.plugins.fightcave;
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/JadOverlay.java]: Copyright to the original author

**Comment by Nightfirecat** [comment]
Needs rebase. Also, I can say personally I don't mind the prayer overlay, but I wouldn't want this spawn overlay.

**Comment by raiyni** [comment]
This needs more than rebasing.

**Pull request closed without merging**